### ISSUE TITLE
Several fonts at the same time for different elements of the code

### ISSUE THREAD

**Issue by danielmunoz** [Microsoft/vscode#23707]
Please create a way to give themes and/or users the possibility of using several fonts at the same time. Please check this article that describes how this is possible to achieve on the Atom editor.

https://medium.com/@peterpme/operator-mono-fira-code-the-best-of-both-worlds-191be512fd5e

In this example, this allows the user the define one font for certain elements in the code, while using another fonts as the main one for the rest of the code.

> If Operator Mono and Fira Code spent a lot of time together and had a baby, you’d get [something that looks like this](https://cdn-images-1.medium.com/max/800/1*hE_nLB776KUDXPERE_3cXw.png).

**Comment by OneOfOne** [comment]
Nice feature request but dear god that's ugly...

**Comment by bcherny** [comment]
Another use case: It's much easier to author Markdown files if the main text uses a Serif font, but that means fenced code blocks also show up in Serif. The ability to have text in Serif and code blocks in Monospace would make authoring Markdown a really nice experience.

**Comment by joefraley** [comment]
this is the thing that i miss most from using Atom. with access to the editor stylesheet i could target operator symbols with fira to get ligatures, and have everything else in operator.

**Comment by thundernixon** [comment]
Just adding that I too would love to use this for editing markdown, or for using a specific variation of a font for certain keywords (e.g. Operator Italic for words like "const" while I use something else for general code). I would just use Atom, but I like VS Code too much to switch.

**Comment by thedonquixotic** [comment]
I would like official support for this. Though I've seen people get a hack around it. Not quite sure how to do it though. Trying to figure that out. They're using custom CSS extension though.

**Comment by nealot** [comment]
Also related: https://github.com/Microsoft/vscode/issues/30925

Would love for this to be supported. Having to rely on extension hacks really sucks.

**Comment by thedonquixotic** [comment]
@nealot check out the  ligaturizer github repo. It's a python script that allows you to add fira code ligatures to any font you want.

Also 8 figured out how to do the custom css thing and it's not hard but all the tutorials on it are way too focused on aping a specific function of a pricy coding font instead of actually giving you just the basic gist. Also really misses the scope of all the things possible with the ability to change fonts. I've got 4 different fonts being used for different aspects of different languages. It's dope!

**Comment by TheSETJ** [comment]
It will be nice if we have this feature, especially with current `tokenColorCustomizations` setting, if we could specify font family too, it'll be great.

**Comment by TheSETJ** [comment]
The problem with class names using `Custom CSS and JS Loader` and extension hacks is that classes are based on styles (it's normal because it's about style sheets) and if various elements on a theme have the same style (for example operators and keywords in my theme are the same color and font style), they have the same class. If you want them to be different you should use `tokenColorCustomizations` to make a difference between them so they get different classes. It would be easier to do other customization on the look of an element using `tokenColorCustomizations`.

**Comment by thedonquixotic** [comment]
Yes espc because of the way that vscode is built. It's my understanding
that Atom doesn't have that issue and that it uses CSS class names but I
could also be wrong. I know that when I asked about the naming scheme for
css class names in vscode I was told they are subject to constant change
and are generated programatically at run time or something of the sort.
Which at least from my observation seems to be true.

**Comment by TheSETJ** [comment]
I confirm that. The behaviour as far as I've seen is as follows: Whenever you change your theme many of the classes regenerate and if you add a rule to the `tokenColorCustomization`, a new `mtkX` class is generated (which X is a number). If you have styles for `mtkY` classes which Y > X, you should find those Ys again and modify your CSS, for classes which Y < X, they stay the same.

**Comment by TheSETJ** [comment]
I want to know, can we expect this feature in any future version? I see that this issue is added in backlog, but @alexandrudima removed their assignment and this issue is assigned to no one now.

**Comment by jcklpe** [comment]
Yes, I would also be interested in hearing. I'm also having an issue with my custom font in comments as can be seen in this issue here: https://github.com/Microsoft/vscode/issues/59587

**Comment by mikemaccana** [comment]
A few common use cases:
 
 - Emphasizing interesting / unusual / large structures. Eg, a function assignment is more complex than a string assignment.
 - Having code in a monospace font, but prose (like comments) in a regular font
 - Focusing on code rather than comments (or vice versa) by controlling font weight

**Comment by borekb** [comment]
For Markdown, [Caret](https://caret.io/) does this beautifully:

_Light:_

<img width="1027" alt="Screenshot 2019-10-23 at 08 55 28" src="https://user-images.githubusercontent.com/101152/67365965-eba44900-f572-11e9-91ec-e88c0b4e50ac.png">

_Dark:_

<img width="1022" alt="Screenshot 2019-10-23 at 08 56 26" src="https://user-images.githubusercontent.com/101152/67366042-08408100-f573-11e9-912e-d1c4e12caadc.png">

I wish this was possible in VSCode.

**Comment by jcklpe** [comment]
@borekb it kind of is with a hack using CSS injection as I mentioned. If you'd like to see how I did it, I have a post [[here]](https://www.jackalope.tech/improve-the-visual-semantics-of-vscode/)

Warning the extension I originally made does not actually work in solving this problem because I packaged the extension wrong (or something... not sure... I used MS's azure based extension build tool and the documentation for it seems outdated so maybe that's just broken). 

I'm going to do a proper rebuild of the project soon but for now, the blog post and [the code in the git-repo](https://github.com/jcklpe/injectedTheme-vscode/tree/2f91fa300e4df2da2171372e2ee8e589892a403a) for the project should give you a basic idea of how this issue can be solved.

**Comment by dziku86** [comment]
@jcklpe Hi, could you re-post [this](https://www.jackalope.tech/improve-the-visual-semantics-of-vscode/) post on your blog?

**Comment by jcklpe** [comment]
@dziku86 Yeah sorry about that. The code for the project has a copy of that post in it's readme: https://github.com/jcklpe/vscode-enzo

**Comment by dziku86** [comment]
> @dziku86 Yeah sorry about that. The code for the project has a copy of that post in it's readme: https://github.com/jcklpe/vscode-enzo

Thank you!
